lib/tpm/DeviceStateHandler.cpp

```diff
@@ -26,6 +26,7 @@ using namespace PAL;
 void DeviceStateHandler::Start()
 {
 	// TRACE("_RetrieveAndRegisterForDeviceConditionChange");
+	std::lock_guard<std::mutex> lock(m_lock);
```

Comment:
I generally don't have a good feeling about adding a mutex guard for the `potential` critical region just by going through the ThreadSanitizer report. What makes you think that the methods within this class would get called from multiple threads simultaneously?

Reply:
My thought was this `Start` probably won't need to be guarded by the mutex, while the `Stop` and `OnChanged` might be needed, what do you think?